eox_nelp/stats/metrics.py

```diff
@@ -0,0 +1,136 @@
+"""eox-nelp Metrics file.
+
+functions:
+    get_cached_courses: Return visible courses.
+    get_course_metrics: Return the metric for the given course_key.
+    get_learners_metric: Return number of learners, for the visible courses.
+    get_instructors_metric: Return number of instructors, for the visible courses.
+    get_courses_metrics: Return metrics for the visible courses.
+"""
+from django.conf import settings
+
+from eox_nelp.edxapp_wrapper.branding import get_visible_courses
+from eox_nelp.edxapp_wrapper.modulestore import modulestore
+from eox_nelp.edxapp_wrapper.site_configuration import configuration_helpers
+from eox_nelp.edxapp_wrapper.student import CourseAccessRole, CourseEnrollment
+from eox_nelp.stats.decorators import cache_method
+
+
+@cache_method
+def get_cached_courses(tenant):  # pylint: disable=unused-argument
+    """
+    Returns the visible courses. This method is just a wrapper of get_visible_courses that
+    allows to cache the result.
+
+    Args:
+        tenant<str>: String tenant identifier(site.domain)
+
+    Return:
+        list[<CourseOverview>]: List of CourseOverview records.
+    """
+    return get_visible_courses()
+
+
+@cache_method
+def get_course_metrics(course_key):
+    """
+    This allows to get the course stats metrics based on the course key.
+
+    Args:
+        course_key<opaque-key>: Course identifier.
+
+    Return:
+        <Dictionary>: Contains the course's metrics.
+    """
+    stats_settings = getattr(settings, "STATS_SETTINGS", {})
+    course = modulestore().get_course(course_key)
+    chapters = course.get_children()
+
+    sequentials = []
+
+    for chapter in chapters:
+        sequentials += chapter.get_children()
+
+    verticals = []
+
+    for sequential in sequentials:
+        verticals += sequential.get_children()
+
+    allowed_block_types = stats_settings.get("ALLOWED_VERTICAL_BLOCK_TYPES", [])
+    components = {key: 0 for key in allowed_block_types}
```

Comment:
To analize a tenant I have the problem here that you must control the tenant xblocks installed, ok this is could be a general setting for sagar but this set that we have to managed(know) the installed xblocks. Something that is interesting, but what happen if a new release of openedx come with a new xblock. We have to add this one and add it to the setting?
Or this forces to query one time mongo and get all the xblock types and configure correctly the setting `ALLOWED_VERTICAL_BLOCK_TYPES`.🤔

Reply:
you don't have to control any xblock at all, you have to control the API content that will be shown, so if you have a new xblock and you want to show that metric, just add it to the setting

Reply:
I like that structure however if we just require the total, add a dictionary looks unnecessary and just add more complexity to the response, so do we need any extra information ?

Reply:
I don't like adding the dictionary but it reduces the good shape due to the  "uncertainty" of recovering the xblocks without knowing them. I mean this is an approach to avoid the setting of `ALLOWED_VERTICAL_BLOCK_TYPES`.

Reply:
That setting aims to keep the same kind components in the API response no matter the course content, so the API structure is constant in the same tenant, if I implement this I'm just bringing again the initial problem but in a nested way that is more complex. IMO the discussion shouldn't be how hide the inconsistent structure, if you don't like the setting purpose, tell me arguments of why we should implement an API that change its structure based on the course content

Reply:
I think that the reason is that is a stats API. Stats are based on the data of the app. In this case,  mongo. Mongo is a not relational database so you could set there whatever "kind" of documents. IMO it makes sense that the API should reflect the data stored in Mongo, and in this case, the API reflects the distinct cases of xblocks_types stored. Anyway, this discussion seems.
Course managers have often presented the question of which kind of xblocks we have. 
At the moment there are queries to reach it, but this seems like an opportunity to improve that.
This course's endpoint is not relevant to the tenant stats of the home page, so we could discuss this furthermore in the future.
https://3.basecamp.com/3966315/buckets/12732851/messages/4886008049

Reply:
In a good architecture the implementation shouldn't depend on the source of information, that could be mongo, mysql, postgres, .. etc, so the mongo argument doesn't fit here since this a different layer however you make me think that I should implement something stricter, probably in the future. Finally looks like you are thinking in something like GraphQL that is different to an API REST